app/experimenter/experiments/tests/test_models.py

```diff
@@ -1375,6 +1376,23 @@ def test_allocate_buckets_creates_new_bucket_range_if_population_changes(self):
         experiment.allocate_bucket_range()
         self.assertEqual(experiment.bucket_range.count, 2000)
 
+    def test_allocate_buckets_for_live_approved_rollout(self):
```

Comment:
@jaredlockhart here

Reply:
Does this test that condition though?

Reply:
```
                is_rollout=True,
                status=NimbusConstants.Status.LIVE,
                status_next=NimbusConstants.Status.LIVE,
                publish_status=NimbusConstants.PublishStatus.APPROVED,
```

`NimbusExperimentFactory.Lifecycles.LIVE_APPROVE,`:
```
                status=NimbusConstants.Status.LIVE,
                status_next=NimbusConstants.Status.LIVE,
                publish_status=NimbusConstants.PublishStatus.APPROVED,
```

<img width="682" alt="image" src="https://user-images.githubusercontent.com/43795363/221039108-47ea7be9-c1b5-440f-97c0-ed46a51b9308.png">

Reply:
Right, and how does explicitly calling `experiment.allocate_bucket_range()` cover that line above?

Reply:
Said another way, if you revert the change to `SHOULD_ALLOCATE_BUCKETS` will this model test still pass?

Reply:
true! I'll move this check into the serializer test

Reply:
Experiments are tested in here, so rollouts should be too. I think both tests are valid

Reply:
Right but waht I mean is, if you remove the change to SHOULD_ALLOCATE_BUCKETS, will this test still pass?

Reply:
Yes, but it increases our consistency in the tests. This case wasn't covered here before

Reply:
I would like to keep it

Reply:
Sure, but I'm just trying to encourage you to think about the relationship between changes and test cases.  If this test will pass regardless of the changes in a PR, then we know it's unrelated to the changes in this PR.  If it's covering a case that isn't covered at all, like we see some failure in prod but all tests are passing, that's totally valid, but unrelated to this change.  In general there should be some motivation for adding a test that's related to the expected behaviour of the application, whether it's a new change, a bug fix, etc, and not be re testing things that are already covered by other tests.

Reply:
That's a fair point. In that case a lot of the model & serializer tests may be overlapping their coverage; maybe we should take a look at them

Reply:
Totally, I do that all the time, find the same thing being tested in multiple places and then nuke the duplicates.  Like the theory is, you test all the cases at the lowest level, and then each level above you only test integration.  Of course that gets lost over time so you gotta periodically go back in and remove duplicates.  Also in this particular case it occurred to me afterwards that it's the same as the test above it so we could've also accomplished what I think the spirit of what you were going for with just parameterizing against `is_rollout`.  But anyways as usual, I like tests, tests are good, just good to keep in mind what's being tested where 👍